parser/src/lib.rs

```diff
@@ -24,33 +25,67 @@ pub mod token;
 mod token_reader;
 
 pub async fn parse(text: &str) -> Result<Ast, Error> {
-    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
-    let (normalizer_in, normalizer_out): (Sender<Token>, Receiver<Token>) = mpsc::channel(32);
+    let span = trace_span!("parsing");
+    let _enter = span.enter();
+    debug!(text);
+    let channel = 32;
+    trace!("reading channel {channel:?}");
+    let (lexer_in, lexer_out): (Sender<Token>, Receiver<Token>) =
+        mpsc::channel(channel);
```

Comment:
32 is the capacity, so I think this variable name and the logging is a bit confusing. It sounds like 32 is some channel id, which it's not.